django/db/models/fields/__init__.py

```diff
@@ -1040,7 +1040,8 @@ def _check_max_length_attribute(self, **kwargs):
                     id='fields.E120',
                 )
             ]
-        elif not isinstance(self.max_length, int) or self.max_length <= 0:
+        elif not isinstance(self.max_length, int) or type(self.max_length) == bool \
```

Comment:
Thanks for the so quick review. I'll make the changes.

Yes, there is a track ticket: https://code.djangoproject.com/ticket/28120